src/ArmTemplates/Extractor/EntityExtractors/ApiReleaseExtractor.cs

```diff
@@ -0,0 +1,83 @@
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Constants;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.ApiReleases;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Builders.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.EntityExtractors.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.EntityExtractors
+{
+    public class ApiReleaseExtractor : IApiReleaseExtractor
+    {
+        readonly ILogger<ApiReleaseExtractor> logger;
+        readonly ITemplateBuilder templateBuilder;
+        readonly IApisClient apisClient;
+
+        public ApiReleaseExtractor(
+            ILogger<ApiReleaseExtractor> logger,
+            ITemplateBuilder templateBuilder,
+            IApisClient apisClient
+            )
+        {
+            this.logger = logger;
+            this.templateBuilder = templateBuilder;
+            this.apisClient = apisClient;
+        }
+
+        public Template<ApiReleaseTemplateResources> GenerateSingleApiReleaseTemplate(string apiName, ExtractorParameters extractorParameters)
+        {
+            var apiReleaseTemplate = this.templateBuilder
+                                        .GenerateTemplateWithApimServiceNameProperty()
+                                        .Build<ApiReleaseTemplateResources>();
+
+            var apiReleaseResource = this.GenerateReleaseResource(apiName);
+            apiReleaseTemplate.TypedResources.ApiReleases.Add(apiReleaseResource);
+            
+            return apiReleaseTemplate;
+        }
+
+
+        public async Task<Template<ApiReleaseTemplateResources>> GenerateCurrentApiReleaseTemplate(ExtractorParameters extractorParameters)
+        {
+            var apiReleasesTemplate = this.templateBuilder
+                                        .GenerateTemplateWithApimServiceNameProperty()
+                                        .Build<ApiReleaseTemplateResources>();
+
+            var apis = await this.apisClient.GetAllCurrentAsync(extractorParameters);
+
+            foreach (var api in apis)
+            {
+                var apiReleaseResource = this.GenerateReleaseResource(api.ApiNameWithRevision);
+                apiReleasesTemplate.TypedResources.ApiReleases.Add(apiReleaseResource);
+            }
+            
+            return apiReleasesTemplate;
+        }
+
+        ApiReleaseTemplateResource GenerateReleaseResource(string apiName)
+        {
+            var apiReleaseName = $"{Guid.NewGuid()}";
```

Comment:
why do we supply a Guid here? I don't think that's an error - just asking because it is interesting!

Reply:
Basically, its unique release name for the api. Why GUID: 
https://docs.microsoft.com/en-us/rest/api/apimanagement/current-ga/api-release/list-by-service?tabs=HTTP#code-try-0
If you will fetch all existing releases which were done from UI, there are populated with unique identifier, that was a reason to not make it different